back/src/domain/mutation/login.use-case.ts

```diff
@@ -1,21 +1,27 @@
+import { CryptoService } from "@crypto"
 import { TimeTravellerDataSource } from "@data/source"
 import { LoginInputModel, LoginResponseModel } from "@domain/model"
-import { createToken, generatePasswordWithSalt } from "@security"
-require("dotenv").config()
+import { JwtService } from "@jwt"
+import { Service } from "typedi"
 
+@Service()
 export class LoginUseCase {
-  private readonly repository = new TimeTravellerDataSource()
+  constructor(
+    private readonly repository: TimeTravellerDataSource,
+    private readonly jwtService: JwtService,
+    private readonly cryptoService: CryptoService
+  ) {}
 
   async exec(input: LoginInputModel): Promise<LoginResponseModel> {
     const { passport: passpt, password } = input
     const timeTraveller = await this.repository.findOneByPassport(passpt)
     const salt = timeTraveller?.salt
-    const hashedPassword = generatePasswordWithSalt(
+    const hashedPassword = this.cryptoService.generatePasswordWithSalt(
       password,
       salt ?? "defaultSalt"
     )
 
-    if (timeTraveller?.password !== hashedPassword || !timeTraveller) {
+    if (!timeTraveller || timeTraveller?.password !== hashedPassword) {
```

Comment:
nit: se vc ja testa se o timeTraveller existe na primeira parte da condicao, nao precisa verificar na segunda tb:

```suggestion
    if (!timeTraveller || timeTraveller.password !== hashedPassword) {
```

Outra forma de fazer isso eh a seguinte (mais enxuto):

```suggestion
    if (timeTraveller?.password !== hashedPassword) {
```

como a senha hasheada vai exister, um `undefined !== "hash"` vai dar false se o timeTraveller nao existe. Dessa forma, vc ja testa as duas codicoes numa so